api/hack/ci/ci-upload-gocache.sh

```
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Required for signal propagation to work so
+# the cleanup trap gets executed when the script
+# receives a SIGINT
+set -o monitor
+
+source $(dirname $0)/../lib.sh
+
+if [ -z $GOCACHE_MINIO_ADDRESS ]; then
+  echodate "Fatal: env var GOCACHE_MINIO_ADDRESS unset"
+  exit 1
+fi
+
+GOCACHE_DIR=$(mktemp -d)
+export GOCACHE="${GOCACHE_DIR}"
+export GIT_HEAD_HASH="$(git rev-parse HEAD|tr -d '\n')"
+
+export CGO_ENABLED=0
+cd $(dirname $0)/../..
+
+echodate "Building binaries"
+TEST_NAME="Buil Kubermatic"
+retry 2 make build
```

Are our Go builds flaky or why do we need to retry what should just be a series of `go build` statements? Does Go need an extra round to do things?

---

Primarily because writing out the junit is part of the `retry` func and I thought if I do `retry 1`, someone will remove it later, if its `retry 2`, ppl will assume there is  a reason for that and leave it

---

That seems like an issue easily solved by adding a `# (Ab)using retry for junit handling` comment :-)

---

I actually still like to have it. Worst case the job takes 30 seconds more to fail, bestcase we swallow some transient error